airflow/migrations/versions/0027_2_10_3_fix_dag_schedule_dataset_alias_reference_naming.py

```diff
@@ -39,91 +41,200 @@
+        op.execute(
+            "ALTER TABLE dag_schedule_dataset_alias_reference DROP CONSTRAINT IF EXISTS dsdar_dataset_alias_fkey"
+        )
+        op.execute(
+            "ALTER TABLE dag_schedule_dataset_alias_reference DROP CONSTRAINT IF EXISTS dsdar_dag_id_fkey"
+        )
+        postgres_create_foreignkey_if_not_exists(
+            "dsdar_dataset_fkey",
+            "dag_schedule_dataset_alias_reference",
+            "alias_id",
+            "dataset_alias",
+            "id",
+            op,
+        )
+        postgres_create_foreignkey_if_not_exists(
+            "dsdar_dag_fkey",
+            "dag_schedule_dataset_alias_reference",
+            "alias_id",
+            "dataset_alias",
+            "id",
+            op,
+        )
+    if dialect == "mysql":
+        mysql_drop_foreignkey_if_exists(
+            "dsdar_dataset_alias_fkey", "dag_schedule_dataset_alias_reference", op
+        )
+        mysql_drop_foreignkey_if_exists("dsdar_dag_id_fkey", "dag_schedule_dataset_alias_reference", op)
+        mysql_create_foreignkey_if_not_exists(
+            "dsdar_dataset_fkey",
+            "dag_schedule_dataset_alias_reference",
+            "alias_id",
+            "dataset_alias",
+            "id",
+            op,
+        )
+        mysql_create_foreignkey_if_not_exists(
+            "dsdar_dag_fkey",
+            "dag_schedule_dataset_alias_reference",
+            "alias_id",
+            "dataset_alias",
+            "id",
+            op,
+        )
+    if dialect == "sqlite":
+        op.create_table(
+            "new_table",
+            sa.Column("alias_id", sa.Integer(), primary_key=True, nullable=False),
+            sa.Column("dag_id", sa.String(ID_LEN), primary_key=True, nullable=False),
+            sa.Column("created_at", UtcDateTime(timezone=True), nullable=False),
+            sa.Column("updated_at", UtcDateTime(timezone=True), nullable=False),
+            sa.ForeignKeyConstraint(
+                ("alias_id",),
+                ["dataset_alias.id"],
+                name="dsdar_dataset_fkey",
                 ondelete="CASCADE",
-            )
-
-        if "dsdar_dag_id_fkey" in fk_constraints:
-            _rename_fk_constraint(
-                batch_op=batch_op,
-                original_name="dsdar_dag_id_fkey",
-                new_name="dsdar_dag_fkey",
-                referent_table="dataset_alias",
-                local_cols=["alias_id"],
-                remote_cols=["id"],
+            ),
+            sa.ForeignKeyConstraint(
+                columns=("dag_id",),
+                refcolumns=["dag.dag_id"],
+                name="dsdar_dag_fkey",
                 ondelete="CASCADE",
-            )
+            ),
+            sa.PrimaryKeyConstraint("alias_id", "dag_id", name="dsdar_pkey"),
+        )
+        op.execute(sa.text("INSERT INTO new_table SELECT * FROM dag_schedule_dataset_alias_reference"))
+        op.execute("DROP TABLE dag_schedule_dataset_alias_reference")
```

Comment:
Not saying we absolutely should, but you can technically use alembic for this op

```
op.drop_table("dag_schedule_dataset_alias_reference")
```